BojGuide/11722.cpp

```diff
@@ -0,0 +1,30 @@
+#include <iostream>
+#include <algorithm>
+#include <cstring>
+using namespace std;
+
+int N;
+int seq[1000];
+int dp[1002][1000];
```

Comment:
dp[1001][1000] -> dp[1002][1000]
add some space for dp because default maximum value is needed.